src/Symfony/Component/Lock/Store/MongoDbStore.php

```diff
@@ -143,6 +129,45 @@ public function __construct($mongo, array $options = [], float $initialTtl = 300
         }
     }
 
+    /**
+     * Extract default database and collection from given connection URI and remove collection querystring.
+     *
+     * Non-standard parameters are removed from the URI to improve libmongoc's re-use of connections.
+     *
+     * @see https://www.php.net/manual/en/mongodb.connection-handling.php
+     */
+    private function skimUri(string $uri): string
+    {
+        if (false === $parsedUrl = parse_url($uri)) {
+            throw new InvalidArgumentException(sprintf('The given MongoDB Connection URI "%s" is invalid.', $uri));
+        }
+
+        $query = [];
+        if (isset($parsedUrl['query'])) {
+            parse_str($parsedUrl['query'], $query);
+        }
+
+        if (isset($query['collection'])) {
+            $this->options['collection'] = $query['collection'];
+            $queryStringPos = strrpos($uri, $parsedUrl['query']);
+            unset($query['collection']);
+            $prefix = substr($uri, 0, $queryStringPos);
+            $newQuery = http_build_query($query, '', '&', PHP_QUERY_RFC3986);
+            if (empty($newQuery)) {
+                $prefix = rtrim($prefix, '?');
+            }
+            $suffix = substr($uri, $queryStringPos + \strlen($parsedUrl['query']));
+            $uri = $prefix.$newQuery.$suffix;
```

Comment:
@kralos: I only just came across this PR based on a notification in #37180. I understand what this function is attempting to do, but using `parse_str` may silently corrupting a connection string with repeated `readPreferenceTags` keys (a valid use case, as explained in the [URI options](https://github.com/mongodb/specifications/blob/master/source/uri-options/uri-options.rst) spec). [This comment](https://www.php.net/manual/en/function.parse-str.php#76792) on PHP.net goes into more detail about that.

I think it would be preferable to capture the URI option with a regular expression and then, if anything was found, strip it from the returned string. While collection names have their [own restrictions](https://docs.mongodb.com/manual/reference/limits/#Restriction-on-Collection-Names), for purposes of URI parsing I think it'd be best to use something like `/collection=([^&]*)/i`.

Reply:
see https://github.com/symfony/symfony/issues/37864 / https://github.com/symfony/symfony/pull/37868